src/terminal/input/terminalInput.cpp

```diff
@@ -440,120 +440,150 @@ static bool _translateDefaultMapping(const KeyEvent& keyEvent,
 
-        // Only need to handle key down. See raw key handler (see RawReadWaitRoutine in stream.cpp)
-        if (keyEvent.IsKeyDown())
+    // Only need to handle key down. See raw key handler (see RawReadWaitRoutine in stream.cpp)
+    if (!keyEvent.IsKeyDown())
+    {
+        return false;
+    }
+
+    // Many keyboard layouts have an AltGr key, which makes widely used characters accessible.
+    // For instance on a German keyboard layout "[" is written by pressing AltGr+8.
+    // Furthermore Ctrl+Alt is traditionally treated as an alternative way to AltGr by Windows.
+    // When AltGr is pressed, the caller needs to make sure to send us a pretranslated character in GetCharData().
+    // --> Strip out the AltGr flags, in order for us to not step into the Alt/Ctrl conditions below.
+    if (keyEvent.IsAltGrPressed())
+    {
+        keyEvent.DeactivateModifierKey(ModifierKeyState::LeftCtrl);
+        keyEvent.DeactivateModifierKey(ModifierKeyState::RightAlt);
+    }
+
+    // The Alt modifier initiates a so called "escape sequence".
+    // See: https://en.wikipedia.org/wiki/ANSI_escape_code#Escape_sequences
+    // See: ECMA-48, section 5.3, http://www.ecma-international.org/publications/standards/Ecma-048.htm
+    //
+    // This section in particular handles Alt+Ctrl combinations though.
+    // The Ctrl modifier causes all of the char code's bits except
+    // for the 5 least significant ones to be zeroed out.
+    if (keyEvent.IsAltPressed() && keyEvent.IsCtrlPressed())
+    {
+        auto ch = keyEvent.GetCharData();
+        if (ch == UNICODE_NULL)
         {
-            // For AltGr enabled keyboards, the Windows system will
-            // emit Left Ctrl + Right Alt as the modifier keys and
-            // will have pretranslated the UnicodeChar to the proper
-            // alternative value.
-            // Through testing with Ubuntu, PuTTY, and Emacs for
-            // Windows, it was discovered that any instance of Left
-            // Ctrl + Right Alt will strip out those two modifiers and
-            // send the unicode value straight through to the system.
-            // Holding additional modifiers in addition to Left Ctrl +
-            // Right Alt will then light those modifiers up again for
-            // the unicode value.
-            // Therefore to handle AltGr properly, our first step
-            // needs to be to check if both Left Ctrl + Right Alt are
-            // pressed...
-            // ... and if they are both pressed, strip them out of the control key state.
-            if (keyEvent.IsAltGrPressed())
-            {
-                keyEvent.DeactivateModifierKey(ModifierKeyState::LeftCtrl);
-                keyEvent.DeactivateModifierKey(ModifierKeyState::RightAlt);
-            }
+            // For Alt+Ctrl+Key messages GetCharData() returns 0.
+            // -> Get the char from the virtual key.
+            ch = LOWORD(MapVirtualKeyW(keyEvent.GetVirtualKeyCode(), MAPVK_VK_TO_CHAR));

```

Comment:
@lhecker this may be broken for non-US layouts.
See https://stackoverflow.com/q/72464583/1795050